### PR TITLE
Allow accelerator to instantiate the device

### PR DESCRIPTION
**Pull request by nelyahu**
when instantiating torch.device for HPU it cannot be fed with HPU:1 annotation, but only "HPU".
moving the logic to accelerator will allow to solve this issue, with single line change.

### CODE REVIEW

**Review by tjruwase — APPROVED** [review]

File: deepspeed/runtime/engine.py [hunk 1]
```diff
@@ -977,13 +977,13 @@ def _set_distributed_vars(self, args):
...
         if device_rank >= 0:
             get_accelerator().set_device(device_rank)
-            self.device = torch.device(get_accelerator().device_name(), device_rank)
+            self.device = torch.device(get_accelerator().device_name(device_rank))
             self.world_size = dist.get_world_size()
             self.global_rank = dist.get_rank()
         else:
             self.world_size = 1
             self.global_rank = 0
-            self.device = torch.device(get_accelerator().device_name())
```
Comment (loadams) [inline deepspeed/runtime/engine.py:986]: FYI @nelyahu - the CI is working again, the error seems to be caused by these changes, can you take a look, please?

Comment (loadams) [reply deepspeed/runtime/engine.py:986]: Thanks!

**Comment by loadams** [comment]
Will merge #5275 first, that way we can have the newly added unit tests run on this, then we can merge it?

**Comment by nelyahu** [comment]
@loadams Sure, thanks :)

**Comment by loadams** [comment]
> @loadams Sure, thanks :)

Done now, so the tests should run on this PR, then we can merge it, thanks!

**Comment by nelyahu** [comment]
Halting this PR till habana version 1.16.0 will be released.

**Comment by loadams** [comment]
> Halting this PR till habana version 1.16.0 will be released. CI issue should be fixed there.

Thanks @nelyahu - when it is released, go ahead and just update it in this PR?  Also curious if you can share what is required in that release that is causing failures here since it appears the tests just hang now?

**Comment by nelyahu** [comment]
> > Halting this PR till habana version 1.16.0 will be released. CI issue should be fixed there.
> 
> Thanks @nelyahu - when it is released, go ahead and just update it in this PR? Also curious if you can share what is required in that release that is causing failures here since it appears the tests just hang now?
> 
> And just tag one of us when it is updated and we can get this merged.

Sure @loadams , i will update.
I verified this patch on v1.16.0, as our dev process is on the next release branch, and i did not verified v1.14.0 (the release which is being tested here). There was a lot of bug fixed since then, cannot say for sure what is the root cause.

**Comment by loadams** [comment]
> > > Halting this PR till habana version 1.16.0 will be released. CI issue should be fixed there.
> > 
> > 
> > Thanks @nelyahu - when it is released, go ahead and just update it in this PR? Also curious if you can share what is required in that release that is causing failures here since it appears the tests just hang now?
> > And just tag one of us when it is updated and we can get this merged.
> 
> Sure @loadams , i will update. I verified this patch on v1.16.0, as our dev process is on the next release branch, and i did not verified v1.14.0 (the release which is being tested here). There was a lot of bug fixed since then, cannot say for sure what is the root cause.

Thanks @nelyahu - feel free to tag us when it the new version is released and updated so we can merge this then, thanks!

**Comment by loadams** [comment]
> > > > Halting this PR till habana version 1.16.0 will be released. CI issue should be fixed there.
> > > 
> > > 
> > > Thanks @nelyahu - when it is released, go ahead and just update it in this PR? Also curious if you can share what is required in that release that is causing failures here since it appears the tests just hang now?
> > > And just tag one of us when it is updated and we can get this merged.
> > 
> > 
> > Sure @loadams , i will update. I verified this patch on v1.16.0, as our dev process is on the next release branch, and i did not verified v1.14.0 (the release which is being tested here). There was a lot of bug fixed since then, cannot say for sure what is the root cause.
> 
> Thanks @nelyahu - feel free to tag us when it the new version is released and updated so we can merge this then, thanks!

@nelyahu - I tried 1.15.1, looks like same errors, so we will wait for 1.16

**Comment by nelyahu** [comment]
> > > > > Halting this PR till habana version 1.16.0 will be released. CI issue should be fixed there.
> > > > 
> > > > 
> > > > Thanks @nelyahu - when it is released, go ahead and just update it in this PR? Also curious if you can share what is required in that release that is causing failures here since it appears the tests just hang now?
> > > > And just tag one of us when it is updated and we can get this merged.
> > > 
> > > 
> > > Sure @loadams , i will update. I verified this patch on v1.16.0, as our dev process is on the next release branch, and i did not verified v1.14.0 (the release which is being tested here). There was a lot of bug fixed since then, cannot say for sure what is the root cause.
> > 
> > 
> > Thanks @nelyahu - feel free to tag us when it the new version is released and updated so we can merge this then, thanks!
> 
> @nelyahu - I tried 1.15.1, looks like same errors, so we will wait for 1.16

@loadams Yes.

**Comment by loadams** [comment]
@nelyahu - now that we have updated the hpu runner to 1.17, should we move ahead with merging this PR?

**Review by loadams — APPROVED** [review]

**Comment by loadams** [comment]
> @nelyahu - now that we have updated the hpu runner to 1.17, should we move ahead with merging this PR?

@nelyahu/ @BacharL - looks like there is an error with this, could you take a look when you have time?

**Comment by nelyahu** [comment]
> > @nelyahu - now that we have updated the hpu runner to 1.17, should we move ahead with merging this PR?
> 
> @nelyahu/ @BacharL - looks like there is an error with this, could you take a look when you have time?

Yes, this test is disabled locally in our env and fails for the same reason. I removed it from gaudi2 workflow file. once fixed will be re-added

**Review by loadams — APPROVED** [review]

**Pull request merged**